File: dotnet/MetalTest/ViewController.cs
using System;
using AppKit;
using CoreGraphics;
using Metal;
using MetalKit;

namespace MetalTest
{
    public class ViewController : NSViewController
    {
        private readonly MTKView _view;
        private readonly Renderer _renderer;

        public ViewController()
        {
            var device = MTLDevice.SystemDefault;
            if (device == null)
            {
                throw new Exception("Metal is not supported on this device");
            }
            var view = new MTKView(new CGRect(), device);
            var renderer = new Renderer(view);
            view.Delegate = renderer;

            _view = view;
            _renderer = renderer;
            base.View = _view;
        }
    }
}

File: mac/Visualizer/Visualizer/ViewController.swift
import Foundation
import Metal
import MetalKit

class ViewController: NSViewController {

    private let mtkView: MTKView
    private let renderer: Renderer

    init() {

        guard let device = MTLCreateSystemDefaultDevice() else {
            fatalError("Metal is not supported on this device")
        }
        print("GPU: \(device)")

        self.mtkView = MTKView(frame: CGRect(), device: device)
        self.renderer = Renderer(view: mtkView)
        mtkView.delegate = renderer
        super.init(nibName: nil, bundle: nil)
        self.view = mtkView
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}